src/notification-service.ts

```diff
@@ -0,0 +1,39 @@
+import * as _ from 'lodash';
+const webpush = require('web-push');
+
+export class NotificationService {
+  private subscriptions: PushSubscriptionJSON[] = [];
+
+  constructor() {
+    const vapidKeys = {
+      publicKey: 'BGk2Rx3DEjXdRv9qP8aKrypFoNjISAZ54l-3V05xpPOV-5ZQJvVH9OB9Rz5Ug7H_qH6CEr40f4Pi3DpjzYLbfCA',
+      privateKey: 'D-k70ba0x5ucasrJMfsROWq8Xtt2smbzh98mbXTfhQM'
+    }
+
+    webpush.setVapidDetails(
+      'mailto:team@angular-buch.com',
+      vapidKeys.publicKey,
+      vapidKeys.privateKey
+    );
+  }
+
+  addSubscription(subscription: PushSubscriptionJSON) {
```

Comment:
Where do those types come from, are they native?

Reply:
jep, they are. But I had to update to typescript 3.5, as the compiler didn't know `PushSubscriptionJSON` before.

Reply:
Hier nen Link: https://github.com/microsoft/TypeScript/blame/aff7ef12305de00cca7ac405fdaf8402ba0e6973/lib/lib.dom.d.ts#L1025 ;-)

Reply:
Okay, hab es jetzt noch etwas eleganter gelöst: e7d4781c1eebf3b5ab74688115474010ff7f3555